### PR TITLE
Adding support for St7796 displays

### LINKED ISSUE #1187: ST7796 4.0" Display Support
**Describe the problem you have/What new integration you would like**

So I have a 4.0" TFT Touch Display based on the ST7796 Driver which supports touch and runs on the SPI bus.
Is there any way to get support for this device in ESPHome?

**Please describe your use case for this integration and alternatives you've tried:**

I have tried the ILI9341 display configuration for this device and I get information on the screen but the size is wrong and it doesn't fill the whole screen.  If I put a 2.4" ILI9341 device on, its perfect but I want a bigger display.

**Additional context**

### PR DESCRIPTION
**Pull request by gpambrozio**
# What does this implement/fix?

Adds support for St7796 displays. This display is used in [Seed Studio's WT32-SC01](https://www.seeedstudio.com/ESP32-Development-board-WT32-SC01-p-4735.html) for example.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1187

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2329

## Test Environment

- [X] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:
```yaml
# Example config.yaml
esphome:
  name: touch_panel

esp32:
  board: m5stack-core2
  framework:
    type: arduino

logger:

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

external_components:
  - source: github://gpambrozio/esphome@ST7796
    components: [ ili9341 ]
    refresh: 1min

spi:
  clk_pin: GPIO14
  mosi_pin: GPIO13

qr_code:
  - id: wifi_qr
    ecc: QUARTILE
    value: "https://esphome.com"

display:
  - platform: ili9341
    model: ST7796
    cs_pin: GPIO15
    dc_pin: GPIO21
    reset_pin: GPIO22
    led_pin: GPIO23
    rotation: 270
    lambda: |-
      it.qr_code(10, 10, id(wifi_qr), Color(255,255,255), 5);

```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/ili9341/ili9341_init.h [hunk 1]
```diff
@@ -65,6 +65,26 @@ static const uint8_t PROGMEM INITCMD_TFT[] = {
   0x00                                   // End of list
 };
 
+static const uint8_t PROGMEM INITCMD_ST7796[] = {
```
Comment (gpambrozio) [inline esphome/components/ili9341/ili9341_init.h:68]: These changes are based on work done [here](https://github.com/esphome/feature-requests/issues/1187#issuecomment-1254195657) and just re-organized to fit esphome a bit better. Thanks @clowrey for figuring this out.

**Comment by nielsnl68** [comment]
Hi @gpambrozio i'm working one a reworked version of the ili9341 display driver by renaming it to the ili9xxx, do you mind when i merge your changes to that component?

**Comment by gpambrozio** [comment]
> Hi @gpambrozio i'm working one a reworked version of the ili9341 display driver by renaming it to the ili9xxx, do you mind when i merge your changes to that component?

Sure, should not be hard. Can you send me the PR so I can follow it?

**Comment by nielsnl68** [comment]
> Sure, should not be hard. Can you send me the PR so I can follow it?

Okay, i implement our changes in my updated ili9xxx platform component.
And is now part of my older PR about the ILI9341 from a couple of weeks ago: #3795

**Comment by gpambrozio** [comment]
This should be implemented as part of https://github.com/esphome/esphome/pull/3795 so I'll close this. May re-open later if the other change doesn't make it.

**Pull request closed without merging**